tests/test_unit_propagation.py

```
@@ -0,0 +1,214 @@
+    assert 'joule' in sample[1].lower() or 'pj' in sample[1].lower()
+
+    conn.close()
+
+
+def test_units_propagated_to_capacity_tables(solved_db_with_units: Path) -> None:
+    """Verify that capacity output tables have units populated."""
+    conn = sqlite3.connect(solved_db_with_units)
+
+    for table in ['output_built_capacity', 'output_net_capacity']:
+        row = conn.execute(
+            f'SELECT COUNT(*), SUM(CASE WHEN units IS NOT NULL THEN 1 ELSE 0 END) FROM {table}'
```

Ruff (0.14.8)

Possible SQL injection vector through string-based query construction

(S608)